core/src/com/mygdx/crazybomber/Room.java

```diff
@@ -0,0 +1,145 @@
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+
+public class Room implements Screen {
+    private Sprite background,avatar, avatarReady, avatarNotReady;
+    private SpriteBatch batch;
+    private Stage stage;
+    private Table table;
+    private TextButton buttonBack, buttonReady, buttonStart;
+    private Label heading;
+    private Skin skin;
+    private TextureAtlas atlas;
+    private Texture splashTexture;
+    private MapRepository mapRepository;
+    private StateRepository stateRepository;
+    private boolean ready = false;
+
+    public Room (Texture texture, final StateRepository stateRepository, MapRepository mapRepository) {
+        this.mapRepository = mapRepository;
+        this.splashTexture = texture;
+        this.stateRepository = stateRepository;
+    }
+
+    @Override
+    public void show() {
+        batch = new SpriteBatch();
+        background = new Sprite(splashTexture);
+        background.setSize(Gdx.graphics.getWidth(), Gdx.graphics.getHeight());
+
+        avatarReady = new Sprite(new Texture("ready.png"));
+        avatarReady.setPosition(100,300);
+        avatarReady.setSize(200,200);
+        avatarNotReady = new Sprite(new Texture("notready.png"));
+        avatarNotReady.setPosition(100,300);
+        avatarNotReady.setSize(200,200);
+        avatar = avatarNotReady;
+
+        stage = new Stage();
+        Gdx.input.setInputProcessor(stage);
+        atlas = new TextureAtlas("ui/button.pack");
+        skin = new Skin(Gdx.files.internal("menuSkin.json"),atlas);
+        table = new Table(skin);
+        table.setBounds(0,0,Gdx.graphics.getWidth(),Gdx.graphics.getHeight());
+
+        buttonBack = new TextButton("BACK", skin);
+        buttonBack.addListener(new ClickListener() {
+            @Override
+            public void clicked(InputEvent event, float x, float y) {
+                ((Game) Gdx.app.getApplicationListener()).setScreen(stateRepository.peek());
+            }
+        });
+
+        buttonReady = new TextButton("READY", skin);
+        buttonReady.addListener(new ClickListener() {
+            @Override
+            public void clicked(InputEvent event, float x, float y) {
+                if(ready == false){
+                    ready = true;
+                    avatar = avatarReady;
+                }
+                else {
+                    ready = false;
+                    avatar = avatarNotReady;
+                }
+            }
+        });
+
+        buttonStart = new TextButton("START", skin);
+        buttonStart.addListener(new ClickListener() {
+            @Override
+            public void clicked(InputEvent event, float x, float y) {
+                //TODO: if logic to check if everyone's ready
```

Comment:
So usually if theres a button like this, it disables itself so the users cant even click it until everyone's ready. the Button class should have something like a disable, do that instead of checking if everyone's ready after they click the button

Reply:
i just added an if(ready) inside the statement to make it work like that
not sure if optimal